tests/feature-store/test_steps.py

```diff
@@ -42,3 +42,21 @@ def test_set_event_random_id():
     resp = server.test(body={"data": "123"}, event_id="XYZ")
     server.wait_for_completion()
     assert resp["id"] != "XYZ", "id was not overwritten"
+
+
+def test_set_event_with_spaces_or_hyphens():
```

Comment:
This test should fail when run against development to be a proper regression test.